RegressionModels/LinearRegression/HousePriceEstimation.py

```diff
@@ -2,173 +2,82 @@
-print(replace_map_comp)
-
-Data_updated_replace.replace(replace_map_comp, inplace=True)
-
-labels = Data_updated['HeatingQC'].astype('category').cat.categories.tolist()
-replace_map_comp = {'HeatingQC' : {k: v for k,v in zip(labels,list(range(1,len(labels)+1)))}}
-print(replace_map_comp)
-
-Data_updated_replace.replace(replace_map_comp, inplace=True)
-
-labels = Data_updated['GarageType'].astype('category').cat.categories.tolist()
-replace_map_comp = {'GarageType' : {k: v for k,v in zip(labels,list(range(1,len(labels)+1)))}}
-print(replace_map_comp)
-
-Data_updated_replace.replace(replace_map_comp, inplace=True)
-
-labels = Data_updated['GarageFinish'].astype('category').cat.categories.tolist()
-replace_map_comp = {'GarageFinish' : {k: v for k,v in zip(labels,list(range(1,len(labels)+1)))}}
-print(replace_map_comp)
-
-Data_updated_replace.replace(replace_map_comp, inplace=True)
-
-labels = Data_updated['KitchenQual'].astype('category').cat.categories.tolist()
-replace_map_comp = {'KitchenQual' : {k: v for k,v in zip(labels,list(range(1,len(labels)+1)))}}
-print(replace_map_comp)
-
-Data_updated_replace.replace(replace_map_comp, inplace=True)
-
-Data_updated_replace = Data_updated_replace.dropna()
+#filling nan values with there coloumn mean
+Data0.fillna(Data0.mean(), inplace=True)
 
-# train data set 
-train_data = Data_updated_replace.head(n=1168)
+#filling nan values with there coloumn mean
+Data0.fillna(Data0.mean(), inplace=True)
+
+#deleting null values
+Data0 = Data0.dropna()
+
+#converting cato variables into labels
+cato_columns = Data0.columns[Data0.dtypes == 'object']
+
+# function to convert catogorial variables with numarical variables
+def cat2numeric(col:pd.Series)->None:
+    """
+    Convert catg column values to numeric values using 
+    sklearns LabelEncoder
+    """
+    le = preprocessing.LabelEncoder()
+    num_values = le.fit_transform(col.values)
+    col.replace(col.values,num_values, inplace=True)
+    
+#mapping catogorial columns with catcat2numeric function    
+list(map(cat2numeric, [Data0[x] for x in cato_columns]))
+
+# spliting the data frame into train data and test data 
+train_data =  sklearn.model_selection.train_test_split(Data0)[0]
+test_data = sklearn.model_selection.train_test_split(Data0)[1]
 
 # X and y values to fit model 
 features = [x for x in train_data.columns if x not in ['SalePrice']]
 X = train_data[features]
 y = train_data['SalePrice']
-
 # fitting X and y values 
 reg = LinearRegression().fit(X.values, y.values)
 
 # regression score 
 print(reg.score(X, y))
 
-# test data set 
-test_data = Data_updated_replace[-292:]
-
 # values of X and y for test data 
 features = [x for x in test_data.columns if x not in ['SalePrice']]
 X_test = test_data[features]
 y_test = test_data['SalePrice']
 
 # predicting values of test data
 y_hat = reg.predict(X_test.values)
-print(y_hat)
-print(y_test)
-print(mean_squared_error(y_test.values,y_hat,multioutput='raw_values'))
 
```

Comment:
Create a scatter plot of y and line plot of y_hat in a single plot once the model is trained